lib/logstash/inputs/kafka.rb

```diff
@@ -250,23 +250,25 @@ def thread_runner(logstash_queue, consumer)
         codec_instance = @codec.clone
         while !stop?
           records = consumer.poll(poll_timeout_ms)
-          for record in records do
-            codec_instance.decode(record.value.to_s) do |event|
-              decorate(event)
-              if @decorate_events
-                event.set("[@metadata][kafka][topic]", record.topic)
-                event.set("[@metadata][kafka][consumer_group]", @group_id)
-                event.set("[@metadata][kafka][partition]", record.partition)
-                event.set("[@metadata][kafka][offset]", record.offset)
-                event.set("[@metadata][kafka][key]", record.key)
-                event.set("[@metadata][kafka][timestamp]", record.timestamp)
+          if records.count > 0
```

Comment:
I suggest using `next unless records.count > 0`, this way we don't have to introduce an extra indentation level

Reply:
Thanks for your review. I changed the code according to your comment.